src/store/modules/eth/actions.js

```diff
@@ -18,24 +18,25 @@ function storeEthAddress(context) {
   storeCryptoAddress(context.state.crypto, context.state.address)
 }
 
-const initTransaction = (api, context, ethAddress, amount, increaseFee) => {
+const initTransaction = async (api, context, ethAddress, amount, increaseFee) => {
+  const nonce = await api.getTransactionCount(context.state.address)
+  const gasPrice = await api.getGasPrice()
+
   const transaction = {
     from: context.state.address,
     to: ethAddress,
-    value: utils.toWei(amount)
-    // gas: api.fromDecimal(DEFAULT_ETH_TRANSFER_GAS), // Don't take default value, instead calculate with estimateGas(transactionObject)
-    // gasPrice: context.getters.gasPrice // Set gas price to auto calc. Deprecated after London hardfork
-    // nonce // Let sendTransaction choose it
+    value: BigInt(utils.toWei(amount)),
+    // gasLimit: api.fromDecimal(DEFAULT_ETH_TRANSFER_GAS), // Don't take default value, instead calculate with estimateGas(transactionObject)
+    gasPrice,
+    nonce
   }
 
-  return api.estimateGas(transaction).then((gasLimit) => {
-    gasLimit = increaseFee
-      ? BigNumber(gasLimit).times(INCREASE_FEE_MULTIPLIER).toNumber()
-      : gasLimit
+  const defaultGasLimit = await api.estimateGas(transaction)
```

Comment:
Same here. Can we move the code to base-actions and use both for eth and erc20?

Reply:
I will create a task for refactoring `eth-base`. Now it's hard to understand what is happening here.
I think this function must be moved outside the store.